### PR TITLE
Time Commitment Filter for Campaigns Page

### PR DESCRIPTION
**Pull request by lindsayrmaher**
### What's this PR do?

This pull request adds another new filter option to our campaigns page! We will now allow users to filter by time commitment as well as actions and causes.

### How should this be reviewed?

👀 

Kapture 2021-02-16 at 14 07 27

Medium:
Screen Shot 2021-02-16 at 2 13 23 PM

Small:
Screen Shot 2021-02-16 at 2 13 08 PM

### Any background context you want to provide?

Just want to call out that the double quotes around two of the `timeCommitment` keys are because of a bug with Algolia!

### Relevant tickets

References [Pivotal #173713859](https://www.pivotaltracker.com/story/show/173713859).

### Checklist

- [x] This PR has been added to the relevant Pivotal card.
- [ ] Documentation added for new features/changed endpoints.
- [x] Added screenshots of front-end changes on small, medium, and large screens.
- [ ] Added appropriate feature/unit tests.

### CODE REVIEW

File: resources/assets/components/pages/CampaignsPage/CampaignFilters/TimeFilter/TimeFilter.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  </label>
+);
+
+TimeInput.propTypes = {
+  isChecked: PropTypes.bool,
+  handleSelect: PropTypes.func.isRequired,
+  timeCommitmentName: PropTypes.string.isRequired,
+  timeCommitmentValue: PropTypes.string.isRequired,
+};
+
```
Comment (aaronschachter) [inline resources/assets/components/pages/CampaignsPage/CampaignFilters/TimeFilter/TimeFilter.js:49]: This sure looks a lot like the [`CauseInput`](https://github.com/DoSomething/phoenix-next/blob/45e94c9c339f576f8e696783d76e7ec5d8c9e66c/resources/assets/components/pages/CampaignsPage/CampaignFilters/CauseFilter/CauseFilter.js#L26-L55) -- what do you think about creating a general `FilterInput` component to DRY? Could the [ActionTypeInput](https://github.com/DoSomething/phoenix-next/blob/master/resources/assets/components/pages/CampaignsPage/CampaignFilters/ActionFilter/ActionTypeInput.js) use it too?

**Review by aaronschachter — COMMENTED** [review]
I tried re-running the last commit to see if it'd fix the failed test, but didn't seem to do the trick... seems like something else not related to these changes at all is awry...

File: resources/assets/components/pages/CampaignsPage/CampaignFilters/FilterInput.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    <input
+      id={filterValue}
+      checked={isChecked}
+      className="mt-1"
+      name={filterValue}
+      onChange={handleSelect}
+      type="checkbox"
+      value={filterValue}
+    />
+    <span className="pl-4">{filterName}</span>
```
Comment (aaronschachter) [inline resources/assets/components/pages/CampaignsPage/CampaignFilters/FilterInput.js:20]: Nit: line break between siblings
```suggestion

    <span className="pl-4">{filterName}</span>
```

File: resources/assets/components/pages/CampaignsPage/CampaignFilters/TimeFilter/TimeFilter.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    </form>
+  );
+};
+
+TimeFilter.propTypes = {
+  filters: PropTypes.object.isRequired,
+  setFilters: PropTypes.func.isRequired,
+};
+
+export default TimeFilter;
```
Comment (aaronschachter) [inline resources/assets/components/pages/CampaignsPage/CampaignFilters/TimeFilter/TimeFilter.js:85]: Way out of scope for this PR -- but I wonder if there's more abstraction we could do to create a general `CampaignFilter` component that could potentially DRY a lot of these filters, e.g. the Clear All button and form HTML, potentially the handlers...

**Review by aaronschachter — APPROVED** [review]
Sweet! 🍪 🍫

File: resources/assets/components/pages/CampaignsPage/CampaignFilters/TimeFilter/TimeFilter.js (hunk 3 above)
Comment (lindsayrmaher) [reply resources/assets/components/pages/CampaignsPage/CampaignFilters/TimeFilter/TimeFilter.js:85]: ++ was thinking this as well, esp also with the analytics tracking etc. i'm going to throw this into an icebox ticket for cleanup!!

**Pull request merged**